Cesium3DTilesContent/test/TestUpgradeBatchTableToExtStructuralMetadata.cpp

```diff
@@ -2319,6 +2368,97 @@ TEST_CASE("Converts \"Feature Classes\" 3DTILES_batch_table_hierarchy example "
   }
 }
 
+TEST_CASE("Omits value-less properties when converting "
+          "3DTILES_batch_table_hierarchy to EXT_structural_metadata") {
+  Model gltf;
+
+  std::string featureTableJson = R"(
+    {
+      "BATCH_LENGTH": 8
+    }
+  )";
+
+  // "Feature classes" example from the spec:
+  // https://github.com/CesiumGS/3d-tiles/tree/main/extensions/3DTILES_batch_table_hierarchy#feature-classes
+  std::string batchTableJson = R"(
+    {
+      "extensions" : {
+        "3DTILES_batch_table_hierarchy" : {
+          "classes" : [
+            {
+              "name" : "Lamp",
+              "length" : 3,
+              "instances" : {
+                "lampStrength" : [10, 5, 7],
+                "lampColor" : ["yellow", "white", "white"],
+                "missingValues": []
+              }
+            },
+            {
+              "name" : "Car",
+              "length" : 3,
+              "instances" : {
+                "carType" : ["truck", "bus", "sedan"],
+                "carColor" : ["green", "blue", "red"]
+              }
+            },
+            {
+              "name" : "Tree",
+              "length" : 2,
+              "instances" : {
+                "treeHeight" : [10, 15],
+                "treeAge" : [5, 8]
+              }
+            }
+          ],
+          "instancesLength" : 8,
+          "classIds" : [0, 0, 0, 1, 1, 1, 2, 2]
+        }
+      }
+    }
+  )";
+
+  rapidjson::Document featureTableParsed;
+  featureTableParsed.Parse(featureTableJson.data(), featureTableJson.size());
+
+  rapidjson::Document batchTableParsed;
+  batchTableParsed.Parse(batchTableJson.data(), batchTableJson.size());
+
+  auto errors = BatchTableToGltfStructuralMetadata::convertFromB3dm(
+      featureTableParsed,
+      batchTableParsed,
+      gsl::span<const std::byte>(),
+      gltf);
+
+  ExtensionModelExtStructuralMetadata* pExtension =
+      gltf.getExtension<ExtensionModelExtStructuralMetadata>();
+  REQUIRE(pExtension);
+  CHECK(
+      gltf.isExtensionUsed(ExtensionModelExtStructuralMetadata::ExtensionName));
+
+  // Check the schema
+  REQUIRE(pExtension->schema);
+  REQUIRE(pExtension->schema->classes.size() == 1);
+
+  auto firstClassIt = pExtension->schema->classes.begin();
+  CHECK(firstClassIt->first == "default");
+
+  Class& defaultClass = firstClassIt->second;
+  REQUIRE(defaultClass.properties.size() == 7);
+
+  // Check the property table
+  REQUIRE(pExtension->propertyTables.size() == 1);
+  PropertyTable& propertyTable = pExtension->propertyTables[0];
+  CHECK(propertyTable.classProperty == "default");
+
+  // Verify that all property table properties refer to a valid bufferView.
+  for (const std::pair<const std::string, PropertyTableProperty>& pair :
+       propertyTable.properties) {
+    CHECK(pair.second.values >= 0);
+    CHECK(size_t(pair.second.values) < gltf.bufferViews.size());
+  }
+}
```

Comment:
I'd add an explicit check that "missingValues" isn't present, for sanity's sake. (I can do that)